From nobody Tue Jul 25 13:22:32 2000
From: Chmouel Boudjnah <chmouel@mandrakesoft.com>
Subject: RFC: patch and backup files.
To: Mandrake Internal <future@linux-mandrake.com>
Date: 25 Jul 2000 13:15:27 -0700

Hi,

With patch the standard behvavior when creating a new file with
backup (patch -b -z.suffix) is to create it as unreadable.

I don't see any reason why to do this, it break gendiff (i need always
to do find . -name .suffix|xargs chmod +r) and is not really
necessary.

Someone again if i :

--- patch-2.5.4/util.c.chmou	Tue Jul 25 12:59:15 2000
+++ patch-2.5.4/util.c	Tue Jul 25 12:59:21 2000
@@ -118,7 +118,7 @@
 
 	  try_makedirs_errno = ENOENT;
 	  unlink (bakname);
-	  while ((fd = creat (bakname, 0)) < 0)
+	  while ((fd = creat (bakname, 0644)) < 0)
 	    {
 	      if (errno != try_makedirs_errno)
 		pfatal ("Can't create file %s", quotearg (bakname));
--- patch-2.5.4/patch.man.chmou	Tue Jul 25 13:13:43 2000
+++ patch-2.5.4/patch.man	Tue Jul 25 13:13:59 2000
@@ -255,7 +255,7 @@
 That is, when patching a file,
 rename or copy the original instead of removing it.
 When backing up a file that does not exist,
-an empty, unreadable backup file is created
+an empty backup file is created
 as a placeholder to represent the nonexistent file.
 See the
 .B \-V

by default ?

-- 
MandrakeSoft Inc                http://www.mandrakesoft.com
San-Francisco, CA USA                             --Chmouel

